client/my-sites/upgrades/checkout/checkout.jsx

```diff
@@ -108,26 +109,32 @@ const Checkout = React.createClass( {
 	},
 
 	addProductToCart: function() {
-		var planSlug = getUpgradePlanSlugFromPath( this.props.product ),
-			cartItem,
+		const planSlug = getUpgradePlanSlugFromPath( this.props.product );
+
+		let cartItem,
 			cartMeta;
 
 		if ( planSlug ) {
 			cartItem = getCartItemForPlan( planSlug );
 		}
 
-		if ( this.props.product.indexOf( 'theme' ) === 0 ) {
-			cartMeta = this.props.product.split( ':' )[1];
+		if ( startsWith( this.props.product, 'theme' ) ) {
+			cartMeta = this.props.product.split( ':' )[ 1 ];
 			cartItem = themeItem( cartMeta );
 		}
 
+		if ( startsWith( this.props.product, 'domain-mapping' ) ) {
```

Comment:
Not introduced in this PR, but for my own understanding: is there a reason we don't use either a query string or a `/` as the delimiter here? The latter is how we handle passing domains through URLs throughout Calypso and and either would be easier to grok if possible, but I suspect there is an issue with doing that in this case.